src/settings.plugin.js

```diff
@@ -248,64 +245,73 @@ wikiaJWPlayerSettingsPlugin.prototype.updateCurrentQuality = function (data) {
 		}
 	}
 };
+// end quality
+
+// captions
+wikiaJWPlayerSettingsPlugin.prototype.onCaptionsChange = function (event) {
+	var emptyCaptionsClass = 'are-captions-empty';
+
+	clearListElement(this.captionsList);
+
+	// tracks always include "off" item
+	if (this.captionsList && event.tracks.length > 1) {
+		var suitableCaptionsTrack = this.getSuitableCaptionsIndex(
+			this.config.selectedCaptionsLanguage || this.captionLangMap[this.getUserLang()],
+			event.tracks
+		);
+
+		event.tracks.forEach(function (track, index) {
+			var captionItem = document.createElement('li'),
+				normalizedLabel = track.label === 'Off' ? 'No captions' : track.label;
+
+			captionItem.dataset.track = index;
+			captionItem.addEventListener('click', function () {
+				this.player.setCurrentCaptions(index);
+				this.adjustCaptionsPosition();
+				this.close();
+				this.player.trigger('captionsSelected', {
+					selectedLang: track.label
+				});
+			}.bind(this));
+
+			captionItem.appendChild(document.createTextNode(normalizedLabel));
+			this.captionsList.insertBefore(captionItem, this.captionsList.firstElementChild);
+		}, this);
+
+		this.wikiaSettingsElement.classList.remove(emptyCaptionsClass);
+		this.show();
```

Comment:
Why is it needed here?

Reply:
yup, it's set only if https://github.com/Wikia/jwplayer-fandom/pull/19/files#diff-9cdf11e3d80f47214396896bb8eefe50R257 condition is met